shopelectro/views/ecommerce.py

```diff
@@ -34,6 +36,37 @@ def get_context_data(self, request, **kwargs):
         }
 
 
+# @todo #789:60m Make cart routes REST style instead of RPC.
+
+# @todo #789:15m Test http cache for cart-get route.
+
+
+def cart_etag(request):
+    cart = SECart(request.session)
+    positions = ','.join(f'{id}:{pos["quantity"]}' for id, pos in cart)
+    return quote_etag(f'{len(cart)}={positions}')
+
+
+class Cart(ec_views.CartModifier):
+
+    cart = SECart
+    position_model = Product
+    order_form = OrderForm
+
+    def get(self, request):
+        return self.json_response(request)
+
+    @classmethod
+    def as_view(cls, *args, **kwargs):
+        # Prevent django cache middleware to add default max-age
+        # this view should always revalidate a request.
+        force_revalidate = cache_control(max_age=0, must_revalidate=True)
```

Comment:
it seems too complex technique with handy-mixing etags.

This approach should be much more simple as i see:
- an app gives to browser fully cached page with empty cart block
- a browser uses JS to query fresh cart data from a server
- browser renders this data at cart block

Reply:
Yes, you are right, we shouldn't make this complicated logic of cache until it becomes an issue